dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1450,20 +1467,25 @@ class CompileTest {
     }
 
     metrics.addAll(<String, dynamic>{
-      'release_full_compile_millis': watch.elapsedMilliseconds,
+      'release_${deleteGradleCache ? 'initial' : 'full'}_compile_millis': watch.elapsedMilliseconds,
       'release_size_bytes': releaseSizeInBytes,
     });
 
     return metrics;
   }
 
-  static Future<Map<String, dynamic>> _compileDebug({
+  Future<Map<String, dynamic>> _compileDebug({
+    required bool deleteGradleCache,
     required bool clean,
     required String metricKey,
   }) async {
     if (clean) {
       await flutter('clean');
     }
+    if (deleteGradleCache) {
+      final Directory gradleCacheDir = Directory('$testDirectory/android/.gradle');
+      gradleCacheDir.deleteSync(recursive: true);
```

Comment:
You can use `rmTree` when you reland this.
 https://github.com/flutter/flutter/blob/7122a0095cfb2f51f0cf7fceca4257ed802c59dc/dev/devicelab/lib/framework/utils.dart#L98-L114